tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -26,20 +28,391 @@ def test_hyperopt_parameters(use_hyperband: bool) -> None:
+        sampler = TPESampler(n_startup_trials=5, seed=0, multivariate=True)
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        uniform_suggestion = sampler.sample_relative(study, trial, {"param-a": uni_dist})
+    assert 1.0 <= uniform_suggestion["param-a"] < 100.0
+
+
+def test_sample_relative_log_uniform_distributions() -> None:
+    """Prepare sample from uniform distribution for cheking other distributions."""
+
+    study = optuna.create_study()
+
+    uni_dist = optuna.distributions.UniformDistribution(1.0, 100.0)
+    past_trials = [frozen_trial_factory(i, dist=uni_dist) for i in range(1, 8)]
+    trial = frozen_trial_factory(8)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+        sampler = TPESampler(n_startup_trials=5, seed=0, multivariate=True)
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        uniform_suggestion = sampler.sample_relative(study, trial, {"param-a": uni_dist})
+
+    # Test sample from log-uniform is different from uniform.
+    log_dist = optuna.distributions.LogUniformDistribution(1.0, 100.0)
+    past_trials = [frozen_trial_factory(i, dist=log_dist) for i in range(1, 8)]
+    trial = frozen_trial_factory(8)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+        sampler = TPESampler(n_startup_trials=5, seed=0, multivariate=True)
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        loguniform_suggestion = sampler.sample_relative(study, trial, {"param-a": log_dist})
+    assert 1.0 <= loguniform_suggestion["param-a"] < 100.0
+    assert uniform_suggestion["param-a"] != loguniform_suggestion["param-a"]
+
+
+def test_sample_relative_disrete_uniform_distributions() -> None:
+    """Test samples from discrete have expected intervals."""
+
+    study = optuna.create_study()
+    disc_dist = optuna.distributions.DiscreteUniformDistribution(1.0, 100.0, 0.1)
+
+    def value_fn(idx: int) -> float:
+        random.seed(idx)
+        return int(random.random() * 1000) * 0.1
+
+    past_trials = [frozen_trial_factory(i, dist=disc_dist, value_fn=value_fn) for i in range(1, 8)]
+    trial = frozen_trial_factory(8)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", optuna.exceptions.ExperimentalWarning)
+        sampler = TPESampler(n_startup_trials=5, seed=0, multivariate=True)
+    with patch.object(study._storage, "get_all_trials", return_value=past_trials):
+        discrete_uniform_suggestion = sampler.sample_relative(study, trial, {"param-a": disc_dist})
+    assert 1.0 <= discrete_uniform_suggestion["param-a"] <= 100.0
+    assert (
+        abs(
+            int(discrete_uniform_suggestion["param-a"] * 10)
+            - discrete_uniform_suggestion["param-a"] * 10
+        )
+        < 1e-3
+    )
```

Comment:
How about using `np.testing.assert_almost_equal` here?
```suggestion
    np.testing.assert_almost_equal(int(discrete_uniform_suggestion["param-a"] * 10), discrete_uniform_suggestion["param-a"] * 10)
```